### PR TITLE
Add table of contents to JSON_INFO.md

### PR DESCRIPTION
**Pull request by zachary-kaelan**
#### Summary
```SUMMARY: Infrastructure "Add table of contents to JSON_INFO.md"```

#### Purpose of change
Makes JSON_INFO.md drastically easier to navigate.

#### Describe the solution
Used [markdown-toc](https://ecotrust-canada.github.io/markdown-toc/) to generate it.

#### Describe alternatives you've considered
Adding a "return to top" link under everything, but as I assumed, one can simply use the `Home` key (`Ctrl-Home` in an editor) to return to the top. I added a note saying this.

### CODE REVIEW

File: doc/HOW_YOU_CAN_HELP.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# How You Can Help
```
Comment (kevingranade) [inline doc/HOW_YOU_CAN_HELP.md]: This kind of thing goes out of date FAST.
It's a lot more scalable to add a search link for e.g. a particular issue tag, like https://github.com/CleverRaven/Cataclysm-DDA/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22 and then focus some effort on maintaining that tagging.

Comment (zachary-kaelan) [reply doc/HOW_YOU_CAN_HELP.md]: I see your point. If it's in a self-contained issue, then the details can be a lot more expansive, and people can comment on it and reference it in commits.

I think instead, we should have a little guide on how to search for these tags, as well as a list of the various tags.

Comment (zachary-kaelan) [reply doc/HOW_YOU_CAN_HELP.md]: What are your thoughts on the new one?

Comment (kevingranade) [reply doc/HOW_YOU_CAN_HELP.md]: Wonderful :+1:

**Pull request merged**